add new tutorial for synthetic biology: basic assembly analysis (#3186)
galaxyproject/training-material: topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md

---
layout: tutorial_hands_on

title: Designing plasmids encoding predicted pathways by using the BASIC assembly method

> zenodo_link: https://zenodo.org/record/6123887#.YhTCPhvjJH4

Reviewer: ```suggestion
zenodo_link: https://zenodo.org/record/6123887
```

Author: Ok, Done.

questions:
- How to design plasmids encoding predicted pathways using BASIC assembly method ?
- How to generate scripts enabling the automatic build of plasmids ?
objectives:
- Annotate the SBML pathway using Selenzy tool.
- Design plasmids encoding predicted pathways using rpbasicdesign tool.
- Generate dnabot scripts enabling the automatic build of plasmids.

> time_estimation: 15 min

Reviewer: ```suggestion
time_estimation: 15M
```

Author: Ok, done.

key_points:
- Annotation of SBML pathway using Selenzyme
- Generating Constructs using rpbasicdesign
- Generating python scripts using dnabot
contributors:
- Kenza BAZI KABBAJ
- Thomas DUIGOU
- Joan HERISSON

> - Guillaume GRICOURT

Reviewer: These need to be replaced with the keys you used in the contributors.yaml file, e.g. `kenza12` instead of the name.

Author: Ok, done.

---

# Introduction
{:.no_toc}

Synthetic biology is a novel engineering discipline which requires computational tools for the design of metabolic pathways for the production of chemicals.

> In this tutorial, we will use a set of tools from the **Genetic Design - BASIC Assembly Workflow** (https://galaxy-synbiocad.org) which will enable you to design plasmids implementing metabolic pathways for the bioproduction of lycopene in _E.coli_ using the [BASIC assembly method](https://doi.org/10.1021/sb500356d), and to generate scripts enabling the automatic build of these plasmids using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.

Reviewer: The DOI you used here indicates this might be a good place to use a reference.
```suggestion
In this tutorial, we will use a set of tools from the **Genetic Design - BASIC Assembly Workflow** (https://galaxy-synbiocad.org) which will enable you to design plasmids implementing metabolic pathways for the bioproduction of lycopene in _E.coli_ using the BASIC assembly method (as described in {% cite ... %}), and to generate scripts enabling the automatic build of these plasmids using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.
```

and then using https://doi2bib.org to get the bibtex entry for that DOI and adding it to the tutorial.bib

Author: Ok, done.

- "parts_for_lycopene.csv" file listing the parts to be used (linkers, backbone and promoters) in the constructions.

- Two YAML files providing two examples of settings, i.e. providing the identifiers of the laboratory equipment and the parameters to be used in the Opentrons scripts.

## Get data

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Create a new history for this tutorial named *Genetic Design - BASIC Assembly Analysis*.
> 2. Import the input files from [Zenodo]({{ page.zenodo_link }}):
>
>    ```
>    https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/dnabot_london_settings.yaml
>    https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/dnabot_paris_settings.yaml
>    https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/parts_for_lycopene.csv
>    https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/rp_002_0011.xml
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
> 3. Rename the datasets if you want to shorten the file names.
>
>    {% snippet faqs/galaxy/datasets_rename.md %}
>
{: .hands_on}

# Find enzymes using **Selenzyme**

> At first, a pathway generated by the Pathway Analysis workflows (https://galaxy-synbiocad.org) is provided as an SBML file to the [Selenzyme](https://doi.org/10.1093/bioinformatics/bty065) tool. Selenzyme searches for enzymes corresponding to each reaction of the pathway. It performs a reaction similarity search in the reference reaction database [MetaNetX](https://www.metanetx.org) and outputs an updated SBML file annotated with the enzyme UniProt IDs.

Reviewer: you might use a `{% cite .. %}` here as well

Author: Ok, done.

---
layout: tutorial_hands_on

title: Designing plasmids encoding predicted pathways by using the BASIC assembly method
zenodo_link: https://zenodo.org/record/6123887
questions:
- How to design plasmids encoding predicted pathways using BASIC assembly method ?
- How to generate scripts enabling the automatic build of plasmids ?
objectives:
- Annotate the SBML pathway using Selenzy tool.
- Design plasmids encoding predicted pathways using rpbasicdesign tool.
- Generate dnabot scripts enabling the automatic build of plasmids.
time_estimation: 15M
key_points:
- Annotation of SBML pathway using Selenzyme
- Generating Constructs using rpbasicdesign

> - Generating python scripts using dnabot

Reviewer 2: Could you rephrase these. They should be statements summing up the take-home messages. So ask yourself, if there are 3-5 things I hope the student will have learned from this, what is it?

So formulated not as "Doing task X", but "When doing X, always make sure to Y", or "Doing X is useful because Z" etc, real conclusions. The way these are phrased now, they are more objectives.

Author: Yes, understood, thank you for your explanation. We changed that.

To design plasmids encoding lycopene bioproducing pathways, we will use the BASIC assembly method (as described in {% cite Storch2015 %}) which relies on orthogonal linkers and type IIs restriction enzyme cleavage to provide a robust and accurate assembly of DNA parts into plasmid constructs. From these construct definition, the workflow will generate scripts enabling the automatic build of the plasmids as well as  the transformation of strains using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.

The workflow scheme we will use is shown below. First, we will run the steps of this workflow individually so as not to neglect the understanding of the intermediate steps as well. Then, we will run the workflow automatically so that it itself retrieves the outputs from the previous step and gives them as input to the next tool.

![Genetic Design - BASIC Assembly Workflow](../../images/basic_assembly_workflow.png)

> Note that a video tutorial is also available for this workflow : [Galaxy BASIC Assembly Workflow Tutorial ](https://youtu.be/it_JTVdvaRo)

Reviewer 2: A video, nice! Would you be ok with adding that to the [GTN Video Library](https://gallantries.github.io/video-library/)? That way the video will be automatically linked from the top of the tutorials, and from the topic page (under "recordings"). And it would also make it easier to create online training events around your tutorials.

We would re-upload this tutorial to the GalaxyProject YouTube account, and ask you to correct the auto-generated captions that YouTube generates, so that we can provide quality subtitles and increase accessibility for our users.  We will copy the autogenerated subs to a Google Doc for you and others to fix, usually its just a matter of punctuation and fixing technical terms)

Author: I submitted the video to the GTN Video Library 2 weeks ago. I don't know if you were able to analyze it ? Thanks.

Author: Hi @shiltemann , thank you for analyzing the video !

Since the video will appear from the top of the tutorial, I deleted the old video link in the tutorial.

About the speaker, no it's not me. He's one of the tutorial contributors:

```
breakthewall:
    name: Joan Hérisson
    email: joan.herisson@univ-evry.fr
    orcid: 0000-0001-9741-0847
    gitter: breakthewall
    linkedin: joanherisson
    joined: 2022-03
```
Name of institute: Université d'Évry / Paris-Saclay, CNRS, CEA, INRAE
Webpage: http://www.univ-evry.fr/

He's an employee of the University of Évry / Paris-Saclay supervised by CNRS and CEA. He's also working in collaboration with INRAE in this project. I will let you pick the needed informations.

Thanks again.

![evry_logo](https://user-images.githubusercontent.com/17780887/168045191-8a49d2d2-6831-4967-aac3-162402c201c6.png)
![Logo-INRAE_Quadri-web](https://user-images.githubusercontent.com/17780887/168045248-6c6e521d-aba6-4732-8b78-f4b248fc7751.jpg)
![CNRS](https://user-images.githubusercontent.com/17780887/168045262-58dd444b-f36e-4ce5-9d9a-fd7a4c45e716.jpg)
![CEA_logo_nouveau](https://user-images.githubusercontent.com/17780887/168045273-dd28eaa9-16f8-4aae-a4d2-beb641fd9018.svg)

Reviewer 2: Perfect, thanks!

Here is a link to the video captions as generated by YouTube, if you could go over it fix where needed, that would be great

https://docs.google.com/document/d/1edwDeJtySnO2d9nLbwPBpoUeBdEYE-9sLKhCJiEJk3k/edit?usp=sharing

Author: Ok, we fixed the video captions, thank you !!

>    https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/rp_002_0011.xml
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
> 3. Rename the datasets if you want to shorten the file names.
>    Rename `rp_002_0011.xml` to `pathway.xml`.
>
>    {% snippet faqs/galaxy/datasets_rename.md %}
>
{: .hands_on}

# Find enzymes using **Selenzyme**

At first, a pathway generated by the **Pathway Analysis workflow** (https://galaxy-synbiocad.org) is provided as input to the Selenzyme tool ({% cite Carbonell2018 %}). Selenzyme searches for enzymes corresponding to each reaction of the pathway. It performs a reaction similarity search in the reference reaction database [MetaNetX](https://www.metanetx.org) and outputs an updated SBML file annotated with the enzyme UniProt IDs.
The tool provides several scores that can be combined in order to define an overall score. Scores are given for reaction similarity, conservation based on a multiple sequence alignment, phylogenetic distance between source organism and host, and additional scores calculated from sequence properties as shown in the example bellow.

![Selenzyme Results](../../images/selenzy_tool.png)

> > ### {% icon hands_on %} Hands-on: Task description

Author: Ok, done, thanks.

>    > ![Dnabot Outputs](../../images/dnabot_outputs.png)
>    > ![BASIC assembly workflow](../../images/BASIC_ASSEMBLY_WF.png)
>    {: .comment}
{: .hands_on}

> ### {% icon question %} Questions
>
> 1. Looking at the scripts names, can you figure out the main steps of DNA-Bot ?
>
> > ### {% icon solution %} Solution
> >
> > 1. DNA-Bot outputs python scripts that implement the 4 assembly steps: clip reactions, purification, assembly and strain transformation. In short, the **Clip reactions** step prepares the mixes for the ligation of the individual DNA parts with the linkers; the **Purification** step purifies the linker-ligated DNA parts using magnetic beads and the Opentrons magnetic module; the **Assembly** step mixes the DNA purified parts to build the final constructs; while the **Transformation** step transforms the chassis micro-organism with the plasmid and inoculates onto agar.
> >
> {: .solution}
>
{: .question}

# Run the **Genetic Design - BASIC Assembly Workflow**

In this section, you can run the Genetic Design - BASIC Assembly Workflow more easily and fastly following these instructions:

> ### {% icon hands_on %} Hands-on: Task description
>

> > 1. Import your **Genetic Design - Basic Assembly Workflow** by uploading the workflow file using the following **URL link** : https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga.

Author: Ok, done, thanks.

>
>    {% snippet faqs/galaxy/workflows_import.md %}
>
> 2. Click on *Workflow* on the top menu bar of Galaxy. You will see **Genetic Design - Basic Assembly Workflow**
> 3. Click on the {% icon workflow-run %} (*Run workflow*) button next to your workflow
> 4. Provide the workflow with the following parameters:
>    - {% icon param-file %} *"Pathway (SBML)":* Select `pathway.xml` from your current history
>    - *"Host taxon ID"*: Leave the default value `83333`. This stands for using E. coli as the chassis host.
>    - *"Comma separated taxon IDs of output enzyme sequences"*: enter `553` which is the taxon ID of the Pantoea ananatis strain from which we want to extact enzymes
>    - {% icon param-file %} *"Linkers and user parts": Select `parts_for_lycopene.csv` from your current history in csv format*
>    - {% icon param-file %} *"DNA-Bot settings (Optionally)"*: Select either `dnabot_paris_settings.yaml` or `dnabot_london_settings.yaml` from your history)
>
>    > ### {% icon comment %} Comment
>    >
>    > All the outputs will be automatically generated and identical to the previous ones. 
>    {: .comment}
>

> 

Author: Ok, done, thanks.
